OpenEdXMobile/src/main/java/org/edx/mobile/social/facebook/GetUserCallback.java

```diff
@@ -0,0 +1,59 @@
+package org.edx.mobile.social.facebook;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.facebook.GraphRequest;
+import com.facebook.GraphResponse;
+
+import org.edx.mobile.social.SocialMember;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class GetUserCallback {
+    public interface IGetUserResponse {
+        void onCompleted(@Nullable SocialMember socialMember);
+    }
+
+    private IGetUserResponse getUserResponse;
+    private GraphRequest.Callback callback;
+
+    public GetUserCallback(final IGetUserResponse getUserResponse) {
+        this.getUserResponse = getUserResponse;
+        callback = new GraphRequest.Callback() {
+            @Override
+            public void onCompleted(GraphResponse response) {
+                SocialMember socialMember = null;
+                try {
+                    JSONObject userObj = response.getJSONObject();
+                    if (userObj == null) {
+                        return;
+                    }
+                    socialMember = jsonToUser(userObj);
+
+                } catch (JSONException e) {
+                    // Handle exception ...
```

Comment:
Should we handle the exception here rather than just having a comment?

Reply:
I will revisit the code block in next pass

Actually the code inspiration had been taken from FB sample https://github.com/facebook/facebook-android-sdk/blob/master/samples/FBLoginSample/src/main/java/com/facebook/fbloginsample/callbacks/GetUserCallback.java#L57

Reply:
I am waiting for Khalid's pass, don't want to do any changes until he is done with his pass.

Reply:
Agreed! This should be handled properly.